appear/src/main/java/com/arcbees/gquery/appear/client/Appear.java

```diff
@@ -42,13 +47,158 @@ public Appear init(GQuery gq) {
+                }
+            }
+        });
+    }
+
+    private static void checkElementAppearance(Element element, AppearImpl impl) {
+
+        if ($(element).isVisible()) {
+            if (isScrolledIntoView(element, impl)) {
+                maybeOnElementAppeared(element, impl);
+            } else {
+                maybeOnElementDisappeared(element, impl);
+            }
+        } else {
+            maybeOnElementDisappeared(element, impl);
+        }
+    }
+
+    private static void maybeOnElementAppeared(Element element, AppearImpl impl) {
+        if (!impl.isAppeared()) {
+            $(element).as(Events).trigger(APPEAR_EVENT);
+            impl.setAppeared(true);
+        }
+    }
+
+    private static void maybeOnElementDisappeared(Element element, AppearImpl impl) {
+        if (impl.isAppeared()) {
+            $(element).as(Events).trigger(DISAPPEAR_EVENT);
+            impl.setAppeared(false);
+        }
+    }
+
+    private static boolean isScrolledIntoView(Element e, AppearImpl impl) {
+        int docViewTop = Window.getScrollTop();
+        int docViewBottom = docViewTop + Window.getClientHeight();
+        int docViewLeft = Window.getScrollLeft();
+        int docViewRight = docViewLeft + Window.getClientWidth();
+
+        AppearOffset offset = impl.getOptions().getOffset();
+        int elemTop = e.getAbsoluteTop() + offset.getTop();
+        int elemLeft = e.getAbsoluteLeft() + offset.getLeft();
+        int elemBottom = elemTop + e.getOffsetHeight() - offset.getBottom();
+        int elemRight = elemLeft + e.getOffsetWidth() - offset.getRight();
+
+        boolean isVerticalVisible = elemTop >= docViewTop && elemBottom <= docViewBottom
+                || elemTop <= docViewBottom && elemBottom > docViewBottom
+                || elemBottom >= docViewTop && elemTop < docViewTop;
+
+        boolean isHorizontalVisible = elemLeft >= docViewLeft && elemRight <= docViewRight
+                || elemLeft <= docViewRight && elemRight > docViewRight
+                || elemRight >= docViewLeft && elemLeft < docViewLeft;
+
+        return isVerticalVisible && isHorizontalVisible;
+    }
+
+    private static Observe createMutationHandler() {
+        if (useMutationObserver) {
+            return $(Document.get()).as(Observe.Observe)
+                    .observe(Observe.createInit()
+                            .childList(true)
+                            .subtree(true), new Function() {
+                        @Override
+                        public void f() {
+                            hasDomMutated = true;
+                        }
+                    });
+        } else {
+            return null;
+        }
+    }
+
+    private static AppearImpl getImpl(Element e) {
+        return impls.get(e);
+    }
 
     protected Appear(GQuery gq) {
         super(gq);
     }
 
+    /**
+     * Force checking the appearance status of selected elements
+     */
+    public Appear checkAppearance() {
```

Comment:
I still don't like it, methods can be used by other methods and then it's just unmanageable to really know where a method should be. Visibility ordering is predictable at the price of being hard to review. While coding it doesn't mater though since we all used our IDE to navigate.